routes/transactionBreakdown.ts

```diff
@@ -0,0 +1,110 @@
+import { Request, Response } from 'express';
+import express from 'express';
+const router = express.Router();
+
+const createItemList = (items: Array<any>) => {
+  return items.map((individual: any, ind: number) => {
+    const indItem = {
+      id: ind,
+      name: individual.name,
+      people: individual.people,
+      cost: individual.cost,
+    };
+    return indItem;
+  });
+};
+
+const getSubAmount = (orderItems: Array<any>, name: string = '') => {
+  let total = 0;
+  for (const item of orderItems) {
```

Comment:
instead of a for loop, we can use :
```
return orderItems.reduce((currTotal, currItem) => {
   accumulator = 0;
   if (name && item.people.includes(name)) {
      accumulator = Math.round(Math.round((item.cost / item.people.length) * 100) / 100);
   }
   return currTotal + accumulator;
}, 0);
```